src/supertux/menu/game_menu.cpp

```diff
@@ -50,13 +53,46 @@ GameMenu::menu_action(MenuItem* item)
       break;
 
     case MNID_RESETLEVEL:
-      MenuManager::instance().clear_menu_stack();
-      GameSession::current()->toggle_pause();
-      GameSession::current()->reset_button = true;
+      if(!g_config->confirmation_dialog)
+      {
+        // instantly reset level
+        MenuManager::instance().clear_menu_stack();
+        GameSession::current()->toggle_pause();
+        GameSession::current()->reset_button = true;
+      }
+      else
+      {
+        // reset confirmation dialog
+        std::unique_ptr<Dialog> dialog(new Dialog);
+        dialog->set_text(_("Are you sure?"));
+        dialog->add_cancel_button(_("No"));
+        dialog->add_default_button(_("Yes"), [] {
+          MenuManager::instance().clear_menu_stack();
+          GameSession::current()->toggle_pause();
+          GameSession::current()->reset_button = true;
+        });
+        MenuManager::instance().set_dialog(std::move(dialog));
+      }
       break;
 
     case MNID_ABORTLEVEL:
-      GameSession::current()->abort_level();
+      if (!g_config->confirmation_dialog)
+      {
+        // instantly exit level
+        GameSession::current()->abort_level();
+      }
+      else
+      {
+        // abort Confirmation Dialog
+        std::unique_ptr<Dialog> dialog(new Dialog);
+        dialog->set_text(_("Do you really want to exit the level?"));
```

Comment:
Same here: As I said, this should probably a uniform `#define` or `static` variable.